docs/network-policy.md

```diff
@@ -0,0 +1,144 @@
+# Cluster Network Policy
+
+ClusterNetworkPolicy is a specification of how workloads within a cluster
+communicate with each other and other external endpoints.
+The ClusterNetworkPolicy is supposed to aid cluster-admins to determine
+the security policy for the cluster, unlike K8s NetworkPolicy, which is
+aimed towards developers to secure their apps.
+
+## The ClusterNetworkPolicy resource
+
+An example ClusterNetworkPolicy might look like this:
+```
+apiVersion: security.antrea.tanzu.vmware.com/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: test-cnp
+spec:
+    priority: 5
+    appliedTo:
+      - podSelector:
+          matchLabels:
+            role: db
+      - namespaceSelector:
+          matchLabels:
+            env: prod
+    ingress:
+      - action: Allow
+        from:
+          - podSelector:
+              matchLabels:
+                role: frontend
+          - podSelector:
+              matchLabels:
+                role: nondb
+            namespaceSelector:
+              matchLabels:
+                role: db
+        ports:
+          - protocol: TCP
+            port: 8080
+    egress:
+      - action: Drop
+        to:
+          - ipBlock:
+            cidr: 10.0.10.0/24
+        ports:
+          - protocol: TCP
+            port: 5978
+```
+
+**spec**: The ClusterNetworkPolicy `spec` has all the information needed to
+define a cluster-wide security policy.
+
+**appliedTo**: The `appliedTo` field specifies the grouping criteria of Pods to
+which the policy applies to. Pods can be selected cluster-wide using
+`podSelector`. If set with a `namespaceSelector`, all Pods from Namespaces
+selected by the namespaceSelector will be selected. Specific Pods from
+specific Namespaces can be selected by using a combination of
+`podSelector` and `namespaceSelector` in the same peer. IPBlock is not allowed
```

Comment:
I'm not sure what a "peer" is in the context of "appliedTo" (as opposed to in the context of rules). Maybe use the following:
> Specific Pods from specific Namespaces can be selected by providing both a `podSelector` and a `namespaceSelector` in the same `appliedTo` entry.

Reply:
maybe `peer` is more technical.. but even in K8s NetworkPolicy a Peer refers to one entry of selectors
for example:
appliedTo:
- pSel1:
- nSel1:
has two peers
while
appliedTo:
- pSel1:
   nSel1:
has only a single peer.
So wanted to clarify that in order to select pods from namespaces, they should be set in a single peer, as opposed to two peers.

Reply:
I think the "peer" terminology only applies to `to` / `from` entries in the rules, not really the workloads to which the policy applies, which is what my comment was about

Reply:
i see your point.. in our CNP .. appliedTo and to/from share the struct Peer.. but it can still be confusing.. updated